code/modules/reagents/chemistry/holder.dm

```diff
@@ -268,7 +280,17 @@
 							var/datum/reagent/A = addiction
 							if(istype(R, A))
 								A.addiction_stage = -15 // you're satisfied for a good while.
-				need_mob_update += R.on_mob_life(C)
+				var/conflicting = FALSE
+				if(conflict_reagents)
+					for(var/assoc_list in conflict_reagents)
+						var/datum/reagent/base_reagent = assoc_list[1]
+						var/datum/reagent/conflict_reagent = assoc_list[2]
+						if(base_reagent == R)
+							conflicting = TRUE
+							if(base_reagent && conflict_reagent)
+								need_mob_update += R.on_reagent_conflict(C, conflict_reagent)
+				if(!conflicting)
+					need_mob_update += R.on_mob_life(C)
```

Comment:
Actually, he's correct. unless I'm wrong. `on_mob_life` is what depletes the reagent, and it's never called if there's conflicting reagents, in lieu of calling ` on_reagent_conflict` instead. So yes, the poisons last forever.

Reply:
yes, it's a bug